src/core/csv/reader_fread.cc

```diff
@@ -296,6 +297,7 @@ void FreadReader::detect_sep_and_qr() {
         if (thisncol < 0) {
           // invalid file with this sep and quote rule; abort
           numFields[0] = -1;
+          if (quoteRule <= 1) check_all_quoteRules = true;
```

Comment:
If parse fails at QR=0 but succeeds at QR=1, then we'll still go into checking all other QRs.
Consider the following test case:
```python
def test_sep_within_quoted_fields3():
    assert_equals(dt.fread('\n"1,\\"2"\n"3,\\"4"\n'),
                  dt.Frame(['1,"2', '3,"4']))
```

Reply:
Changed the logic a little and added this test as well.

Reply:
After submitting the new patch, I realised that the new test I ended up adding, which by the way passes,
```python
def test_sep_within_quoted_fields3():
    assert_equals(dt.fread('cols\n"1,\\"2"\n"3,\\"4"\n'), dt.Frame(cols=['1,"2', '3,"4']))
```
is not the same as what you suggested. From your example, dt.fread('\n"1,\\"2"\n"3,\\"4"\n') gives
```
   | 1,"2
-- + ----
 0 | 3,"4

```
and dt.Frame(['1,"2', '3,"4']) gives
```
   | C0  
-- + ----
 0 | 1,"2
 1 | 3,"4
```
The columns names and the shape of the table don't add up and hence this test would fail on this branch. In fact, this fails on the master branch as well. do you think it's a bug?

Reply:
How about  `dt.fread('"1,\\"2"\n"3,\\"4"\n', header=False)`?
I just don't  want fread to be able to infer that there is only one column solely based on the fact that the first row has only 1 token.

Reply:
We could similarly have a test case with the qr=0 -encoded  quotes:
```
dt.fread('"1,""2"\n"3,""4"\n', header=False)
```